image_tools/CMakeLists.txt

```diff
@@ -13,37 +13,63 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
 find_package(sensor_msgs REQUIRED)
 find_package(std_msgs REQUIRED)
-
+find_package(rcutils REQUIRED)
 find_package(OpenCV REQUIRED)
 
 include_directories(include)
 
-add_executable(cam2image
-  src/burger.cpp
-  src/cam2image.cpp
-  src/options.cpp
-)
-ament_target_dependencies(cam2image
+add_library(showimage_component SHARED
+  src/showimage.cpp)
+ament_target_dependencies(showimage_component
   "rclcpp"
   "sensor_msgs"
-  "std_msgs"
+  "std_msgs"    
+  "rclcpp_components"
   "OpenCV")
+rclcpp_components_register_nodes(showimage_component "image_tools::ShowImage")
 
-add_executable(showimage
-  src/options.cpp
-  src/showimage.cpp
-)
-ament_target_dependencies(showimage
+add_library(cam2image_component SHARED 
+  src/cam2image.cpp)
+ament_target_dependencies(cam2image_component
   "rclcpp"
   "sensor_msgs"
   "std_msgs"
-  "OpenCV")
+  "OpenCV"
+  "rclcpp_components")
+rclcpp_components_register_nodes(cam2image_component "image_tools::Cam2Image")
+
+add_executable(cam2image
+  src/cam2image_main.cpp
+  src/options.cpp
+  src/burger.cpp)
+target_link_libraries(cam2image
+  showimage_component
+  cam2image_component)
+ament_target_dependencies(cam2image
+  "rclcpp")
+
+  add_executable(showimage
+  src/showimage_main.cpp
+  src/options.cpp)
+target_link_libraries(showimage
+  showimage_component
+  cam2image_component)
+ament_target_dependencies(showimage
+  "rclcpp")
+
+install(TARGETS
+  showimage_component
+  cam2image_component
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin)
 
 install(TARGETS
-  cam2image
   showimage
+  cam2image
```

Comment:
nit: This change is unnecessary (I guess they were purposefully alphabetized before).